src/main/java/org/folio/entlinks/controller/delegate/InstanceAuthorityStatServiceDelegate.java

```diff
@@ -0,0 +1,73 @@
+package org.folio.entlinks.controller.delegate;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.folio.entlinks.client.UsersClient;
+import org.folio.entlinks.controller.converter.AuthorityDataStatMapper;
+import org.folio.entlinks.domain.dto.AuthorityChangeStatDtoCollection;
+import org.folio.entlinks.domain.dto.AuthorityDataStatActionDto;
+import org.folio.entlinks.domain.dto.Metadata;
+import org.folio.entlinks.domain.entity.AuthorityDataStat;
+import org.folio.entlinks.service.links.AuthorityDataStatService;
+import org.folio.entlinks.utils.DateUtils;
+import org.folio.spring.tools.model.ResultList;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class InstanceAuthorityStatServiceDelegate {
+
+  private final AuthorityDataStatService dataStatService;
+  private final AuthorityDataStatMapper dataStatMapper;
+  private final UsersClient usersClient;
+
+  public AuthorityChangeStatDtoCollection fetchAuthorityLinksStats(OffsetDateTime fromDate, OffsetDateTime toDate,
+                                                                   AuthorityDataStatActionDto action, Integer limit) {
+    List<AuthorityDataStat> dataStatList = dataStatService.fetchDataStats(fromDate, toDate, action, limit);
+
+    Optional<AuthorityDataStat> last = Optional.empty();
+    if (dataStatList.size() > limit) {
+      last = Optional.of(dataStatList.get(limit));
+    }
+    String query = getUsersQueryString(dataStatList);
+    var userResultList = usersClient.query(query);
+    var stats = dataStatList.stream().limit(limit)
+      .map(source -> {
+        Metadata metadata = getMetadata(userResultList, source);
+        var authorityDataStatDto = dataStatMapper.convertToDto(source);
+        authorityDataStatDto.setMetadata(metadata);
+        return authorityDataStatDto;
+      })
+      .toList();
+
+    return new AuthorityChangeStatDtoCollection()
+      .stats(stats)
+      .next(last.map(authorityDataStat -> DateUtils.fromTimestamp(authorityDataStat.getStartedAt()))
+        .orElse(null));
+  }
+
+  private static Metadata getMetadata(ResultList<UsersClient.User> userResultList, AuthorityDataStat source) {
+    var user = userResultList.getResult()
+      .stream()
+      .filter(u -> UUID.fromString(u.id()).equals(source.getStartedByUserId()))
+      .findFirst().orElseThrow();
```

Comment:
If user was not founded then just not populate user's metadata, don't throw exception